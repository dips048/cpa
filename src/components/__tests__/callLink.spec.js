import React from "react"
import renderer from "react-test-renderer"
import CallLink from "../callLink"
import { render, screen } from "@testing-library/react"


describe("CallLink", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<CallLink/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test("when `Icon` is `true` should renders icon", () => {
    const { getByTestId } = render(<CallLink icon={true} />)
    expect(getByTestId("fa-icon")).toBeInTheDocument()
  })

  test("when `number` is given should renders with given number", () => {
    const { getByText } = render(<CallLink number={"12345676890"} />)
    expect(getByText("12345676890")).toBeInTheDocument()
  })
})
