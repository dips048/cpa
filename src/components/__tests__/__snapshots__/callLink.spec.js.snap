// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallLink renders correctly 1`] = `
<a
  href="tel:(800) 896-4500"
>
  (800) 896-4500
</a>
`;
